jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```diff
@@ -366,9 +411,58 @@ public String getRFC6265SetCookie()
             buf.append("; Secure");
         if (_httpOnly)
             buf.append("; HttpOnly");
+        if (_sameSite != null)
+        {
+            buf.append("; SameSite=");
+            buf.append(_sameSite.toString());
+        }
+
         return buf.toString();
     }
 
+    public static boolean isHttpOnlyInComment(String comment)
+    {
+        return comment != null && comment.contains(HTTP_ONLY_COMMENT);
+    }
+
+    public static SameSite getSameSiteFromComment(String comment)
+    {
+        if (comment != null)
+        {
+            if (comment.contains(SAME_SITE_NONE_COMMENT))
+            {
+                return SameSite.NONE;
+            }
+            if (comment.contains(SAME_SITE_LAX_COMMENT))
+            {
+                return SameSite.LAX;
+            }
+            if (comment.contains(SAME_SITE_STRICT_COMMENT))
+            {
+                return SameSite.STRICT;
+            }
+        }
+
+        return null;
+    }
+
+    public static String getCommentWithoutFlags(String comment)
```

Comment:
Replace "Flags" with "Attributes" - while `HttpOnly` was a boolean "flag" they really are cookie attributes, especially now that `SameSite` is not boolean.